File: Frontend/SoftLibUsuarioBO/ClienteBO.cs
    using SoftLibBO;
using SoftLibBO.ServicioWeb;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SoftLibUsuarioBO
{
    public class ClienteBO : BaseBO
    {
        public int insertar(string nombre, string apellidoPaterno,
                    string apellidoMaterno, string nacionalidad, string numeroDocumento,
                    tipoDocumento tipoDocumento, BindingList<perfil> perfiles, string direccion)
        {
            perfil[] _perfiles = perfiles.ToArray();
            return this.WsCliente.cliente_insertar(nombre, apellidoPaterno,
                   apellidoMaterno, nacionalidad, numeroDocumento,
                     tipoDocumento, _perfiles, direccion);
        }

        public int modificar(int idPersona, string nombre, string apellidoPaterno,
                string apellidoMaterno, string nacionalidad, string numeroDocumento,
                tipoDocumento tipoDocumento, perfil[] perfiles, string direccion)
        {
            perfil[] _perfiles = perfiles.ToArray();
            return this.WsCliente.cliente_modificar(idPersona, nombre, apellidoPaterno,
                 apellidoMaterno, nacionalidad, numeroDocumento,
                 tipoDocumento, perfiles, direccion);
        }

        public int eliminar(int idCliente)
        {
            return this.WsCliente.cliente_eliminar(idCliente);
        }

        public bool existeCliente(int idCliente)
        {
            return this.WsCliente.existeCliente(idCliente);
        }
        public cliente obtenerPorId(int idCliente)
        {
            return this.WsCliente.cliente_obtenerPorId(idCliente);
        }

        public BindingList<cliente> listarTodos()
        {
            cliente[] arreglo = this.WsCliente.cliente_listarTodos();
            return new BindingList<cliente>(arreglo);
        }

        public BindingList<cliente> buscarClientes(string nombre)
        {
            cliente[] arreglo = this.WsCliente.buscarClientes(nombre);
            return new BindingList<cliente>(arreglo);
        }
    }
}

File: Backend/SoftLibUsuarioBO/src/pe/edu/pucp/softlib/usuario/bo/ClienteBO.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package pe.edu.pucp.softlib.usuario.bo;

import java.util.ArrayList;
import pe.edu.pucp.softlib.usuario.dao.ClienteDAO;
import pe.edu.pucp.softlib.usuario.daoImp.ClienteDAOImpl;
import pe.edu.pucp.softlib.usuario.model.Cliente;
import pe.edu.pucp.softlib.usuario.model.Perfil;
import pe.edu.pucp.softlib.usuario.model.TipoDocumento;

/**
 *
 * @author Joshua Haro
 */
public class ClienteBO {
    private final ClienteDAO clienteDAO;
    
    public ClienteBO(){
        this.clienteDAO = new ClienteDAOImpl();
    }
    
    public Integer insertar(String nombre, String apellidoPaterno, 
            String apellidoMaterno, String nacionalidad, String numeroDocumento, 
            TipoDocumento tipoDocumento, ArrayList<Perfil> perfiles,String direccion) {
        Cliente cliente = new Cliente(null, nombre, apellidoPaterno, apellidoMaterno, 
                nacionalidad, numeroDocumento,tipoDocumento,perfiles,direccion);
        return this.clienteDAO.insertar(cliente);
    }
    
    public Integer modificar(Integer idPersona,String nombre, String apellidoPaterno, 
            String apellidoMaterno, String nacionalidad, String numeroDocumento, 
            TipoDocumento tipoDocumento, ArrayList<Perfil> perfiles,String direccion) {
        Cliente cliente = new Cliente(idPersona, nombre, apellidoPaterno, apellidoMaterno, 
                nacionalidad, numeroDocumento,tipoDocumento,perfiles,direccion);
        return this.clienteDAO.modificar(cliente);
    }
    
    public Integer eliminar(Integer idCliente) {
        Cliente cliente = new Cliente();
        cliente.setIdPersona(idCliente);
        return this.clienteDAO.eliminar(cliente);
    }
    
    public ArrayList<Cliente> listarTodos(){
        return this.clienteDAO.listarTodos();
    }
    
    public Cliente obtenerPorId(Integer idCliente){
        return this.clienteDAO.obtenerPorId(idCliente);
    }
    
    public Boolean existeCliente(Integer idCliente){
        Cliente cliente = new Cliente();
        cliente.setIdPersona(idCliente);
        return this.clienteDAO.existeCliente(cliente);
    }
    
    public ArrayList<Cliente> buscarClientes(String nombre){
        return this.clienteDAO.buscarClientes(nombre);
    }
}
